Add tab panels to TiDB Quick Start page and optimize content and structure (#5613)
pingcap/docs: quick-start-with-tidb.md

# Quick Start Guide for the TiDB Database Platform

> This guide walks you through the quickest way to get started with TiDB. You will be using TiUP, a package manager in the TiDB ecosystem, to help you run any TiDB cluster component with only a single line of commands.

Reviewer: ```suggestion
This guide walks you through the quickest way to get started with TiDB. You will be using TiUP, a package manager in the TiDB ecosystem, to help you run any TiDB cluster component with only a single line of command.
```

> If you want to deploy an on-premise production cluster instead, head to [production installation guide](/production-deployment-using-tiup.md). If you want to deploy TiDB in Kubernetes, head to [Get Started with TiDB in Kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/stable/get-started).

Author: ```suggestion
To deploy an on-premise production cluster instead, head to [production installation guide](/production-deployment-using-tiup.md). To deploy TiDB in Kubernetes, head to [Get Started with TiDB in Kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/stable/get-started). To manage TiDB in the cloud, head to [TiDB Cloud Quick Start](https://docs.pingcap.com/tidbcloud/beta/tidb-cloud-quickstart).
```

> To deploy an on-premise production cluster instead, head to [production installation guide](/production-deployment-using-tiup.md). To deploy TiDB in Kubernetes, head to [Get Started with TiDB in Kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/stable/get-started). To manage TiDB in the cloud, head to [TiDB Cloud Quick Start](https://docs.pingcap.com/tidbcloud/beta/tidb-cloud-quickstart).

Reviewer: ```suggestion
To deploy an on-premise production cluster instead, go to [production installation guide](/production-deployment-using-tiup.md). To deploy TiDB in Kubernetes, go to [Get Started with TiDB in Kubernetes](https://docs.pingcap.com/tidb-in-kubernetes/stable/get-started). To manage TiDB in the cloud, go to [TiDB Cloud Quick Start](https://docs.pingcap.com/tidbcloud/beta/tidb-cloud-quickstart).
```

    - If you want to start a TiDB cluster of the latest version with 1 TiDB instance, 1 TiKV instance, 1 PD instance, and 1 TiFlash instance, run the following command:

        {{< copyable "shell-regular" >}}

        ```shell
        tiup playground
        ```

    - If you want to specify the TiDB version and the number of the instances of each component, run a command like this:

        {{< copyable "shell-regular" >}}

        ```shell
        tiup playground v5.0.1 --db 2 --pd 3 --kv 3 --monitor
        ```

        The command downloads a version cluster to the local machine and starts it, such as v5.0.1. `--monitor` means that the monitoring component is also deployed.

        To view the latest version, run `tiup list tidb`.

        This command returns the access methods of the cluster:

        ```log
        CLUSTER START SUCCESSFULLY, Enjoy it ^-^
        To connect TiDB: mysql --host 127.0.0.1 --port 4000 -u root
        To connect TiDB: mysql --host 127.0.0.1 --port 4001 -u root
        To view the dashboard: http://127.0.0.1:2379/dashboard
        To view the monitor: http://127.0.0.1:9090
        ```

        > **Note:**
        >

>         > For the playground operated in this way, after the test is finished, TiUP will clean up the original cluster data. You will get a new cluster after re-running the command.

Reviewer: ```suggestion
        > For the playground operated in this way, after the test deployment is finished, TiUP will clean up the original cluster data. You will get a new cluster after re-running the command.
```

        > If you want the data to be persisted on storage，run `tiup --tag <your-tag> playground ...`. For details, refer to [TiUP Reference Guide](/tiup/tiup-reference.md#-t---tag).

4. Start a new session to access TiDB:

    + Use the TiUP client to connect to TiDBs.

        {{< copyable "shell-regular" >}}

        ```shell

>         tiup client

Reviewer 2: ```
my:root@127.0.0.1:4000=> \conninfo
Connected with driver mysql (root@tcp(127.0.0.1:4000)/?loc=Local&parseTime=true&sql_mode=ansi)
my:root@127.0.0.1:4000=> select "foo";
error: mysql: 1054: Unknown column 'foo' in 'field list'
my:root@127.0.0.1:4000=> select 'foo';
 foo 
-----
 foo 
(1 row)

my:root@127.0.0.1:4000=> select 'foo' || 'bar';
 'foo' || 'bar' 
----------------
 foobar         
(1 row)

my:root@127.0.0.1:4000=>  
```
People probably don't expect an SQL mode that is different than what MySQL (and TiDB) does by default.

Author: This PR aims to solve some specific issues and is large enough for reviewers. To merge this PR quickly, I suggest we keep each PR relatively small. 

For new ideas for improving this page, may I suggest that we discuss and apply them in another new PR? I added a "requires-followup" label to follow up on your suggestions after we merge this PR.

Reviewer 3: That looks like another issue with usql :(

The goal here was to remove a dependency, but it's got some issues. We'll have to ask the tiup team to fix this.

Reviewer 2: Looks like usql is doing this on purpose. Probably to keep the behaviour similar for various DB servers.

https://github.com/xo/usql/blob/ac3c08f3d9cd7e92712479d3898a3b77f08a2c5e/drivers/mysql/mysql.go#L40-L44

>     + Use the TiUP client to connect to TiDBs.

Reviewer: ```suggestion
    + Use the TiUP client to connect to TiDB.
```

        ```

    + You can also use the MySQL client to connect to TiDB.

        {{< copyable "shell-regular" >}}

        ```shell

>         mysql --host 127.0.0.1 --port 4000 -u root

Reviewer 2: Without a `-p` the `mysql` command will use the password from `~/.my.cnf` etc. if that's configured. This may happen if the user is already using MySQL. I suggest to add a `-p` and then have the user just press enter at the password prompt.

I would also suggest to add `--comments` as otherwise output from `SHOW CREATE TABLE...` with TiDB specific comments won't be executed correctly.

We *may* consider using MySQL Shell instead:
```
mysqlsh --sql mysql://root@127.0.0.1:4000
```
instead of
```
mysql --host 127.0.0.1 --port 4000 -u root -p --comments
```
Another option is `tiup client` however as the user may be familiar with the MySQL Client or MySQL Shell this is probably not the best option.

Author: This PR aims to solve some specific issues and is large enough for reviewers. To merge this PR quickly, I suggest we keep each PR relatively small. 

For new ideas for improving this page, may I suggest that we discuss and apply them in another new PR? I added a "requires-followup" label to follow up on your suggestions after we merge this PR.

Reviewer 3: This is the "first start" for absolute beginners. Because `mysql` is available in more distro repos, we should prefer it. It also doesn't need the password option.

Reviewer 2: @yikeke That sounds good to me.

        ```

5. Access the Prometheus dashboard of TiDB at <http://127.0.0.1:9090>.

> 6. Access the [TiDB Dashboard](/dashboard/dashboard-intro.md) at <http://127.0.0.1:2379/dashboard>. The default username is `root`, with an empty password.

Reviewer 2: Maybe we should add Grafana. I think that uses `admin/admin` as default credentials, not sure if that's included in this doc or the `tiup playground` output.

Author: This PR aims to solve some specific issues and is large enough for reviewers. To merge this PR quickly, I suggest we keep each PR relatively small. 

For new ideas for improving this page, may I suggest that we discuss and apply them in another new PR? I added a "requires-followup" label to follow up on your suggestions after we merge this PR.

Reviewer 3: Lets add it later. It doesn't make sense to have the prometheus dashboard and not grafana.

7. (Optional) [Load data to TiFlash](/tiflash/use-tiflash.md) for analysis.

> 8. Clean up the cluster after the test:

Reviewer: ```suggestion
8. Clean up the cluster after the test deployment:
```

    1. Stop the process by pressing `ctrl-c`.

    2. Run the following command:

        {{< copyable "shell-regular" >}}

        ```shell
        tiup clean --all
        ```

> **Note:**
>

> > TiUP Playground listens on `127.0.0.1` by default, and the service is only locally accessible; if you want the service to be externally accessible, specify the listening address using the `--host` parameter to bind the network interface card (NIC) to an externally accessible IP address.

Reviewer: ```suggestion
> TiUP Playground listens on `127.0.0.1` by default, and the service is only locally accessible. If you want the service to be externally accessible, specify the listening address using the `--host` parameter to bind the network interface card (NIC) to an externally accessible IP address.
```

        tiup clean --all
        ```

> **Note:**
>
> TiUP Playground listens on `127.0.0.1` by default, and the service is only locally accessible; if you want the service to be externally accessible, specify the listening address using the `--host` parameter to bind the network interface card (NIC) to an externally accessible IP address.

> > **Note:**
> >
> > TiUP Playground listens on `127.0.0.1` by default, and the service is only locally accessible; if you want the service to be externally accessible, you can specify the listening address using the `--host` parameter to bind the network interface card (NIC) to an externally accessible IP address.

Author: ```suggestion
```

> **Note:**
>
> TiUP Playground listens on `127.0.0.1` by default, and the service is only locally accessible; if you want the service to be externally accessible, specify the listening address using the `--host` parameter to bind the network interface card (NIC) to an externally accessible IP address.

> ## Set up a test environment on a single machine using TiUP cluster

Reviewer 4: Is the `Set up a test environment on a single machine using TiUP cluster` content on the `Deploy a local test environment on Linux OS` tab?

Author: It‘s under "Linux" tab

> **Note:**
>
> TiUP Playground listens on `127.0.0.1` by default, and the service is only locally accessible; if you want the service to be externally accessible, specify the listening address using the `--host` parameter to bind the network interface card (NIC) to an externally accessible IP address.

> 

Author: ```suggestion
```

        tiup cluster display <cluster-name>
        ```

</div>

</SimpleTab>

## What's next

- If you have just deployed a TiDB cluster for the local test environment:

    - Learn [Basic SQL operations in TiDB](/basic-sql-operations.md)
    - [Migrate data to TiDB](/migration-overview.md)

>     - Learn [TiDB key features and scenarios](/overview.md)

Author: ```suggestion
```

>     - Learn [TiDB key features and scenarios](/overview.md)

Author: ```suggestion
```

>     - Learn [TiDB's architecture](/tidb-architecture.md)

Reviewer 4: Why this item is removed?

Author: The architecture is for reference. It's not as important as others. @TomShawn 

And I deleted /overview.md and /mysql-compatibility.md because they are not "what's next". According to the left-hand navigation structure, they are read before the quick start page.

>     - Learn [TiDB's compatibility with MySQL](/mysql-compatibility.md)

Author: ```suggestion
```
